pkg/prometheus_ingester/prometheus_ingester_test.go

```diff
@@ -721,133 +720,134 @@ func Test_processHistogramIncrease(t *testing.T) {
 	}
 }
 
-func TestPrometheusIngesterHttpHeader_UnmarshalYAML(t *testing.T) {
-	const envName = "ENVNAME"
-	var expectedValue = "value"
-	var expectedValueWithPrefix = "xxxvalue"
-
-	if err := os.Unsetenv("NON_EXISTING_ENV_NAME"); err != nil {
-		t.Fatal(err)
-	}
-
-	if err := os.Setenv(envName, expectedValue); err != nil {
-		t.Fatal(err)
-	}
+func Test_httpHeaders_toMap(t *testing.T) {
+	var headerValue = "value"
+	var headerValue2 = "value2"
 
 	tests := []struct {
-		name          string
-		data          []byte
-		expectedValue *string
-		wantErr       bool
+		name    string
+		hs      httpHeaders
+		want    map[string]string
+		wantErr bool
 	}{
 		{
-			name: "value from string",
-			data: []byte(`
-name: name
-value: value
-`),
-			expectedValue: &expectedValue,
-		},
-		{
-			name: "value from env",
-			data: []byte(`
-name: name
-valueFromEnv:
-  name: ENVNAME
-`),
-			expectedValue: &expectedValue,
-		},
-		{
-			name: "value from env with no env name",
-			data: []byte(`
-name: name
-valueFromEnv: {}
-`),
-			wantErr: true,
-		},
-		{
-			name: "value from env with valuePrefix",
-			data: []byte(`
-name: name
-valueFromEnv:
-  name: ENVNAME
-  valuePrefix: xxx
-`),
-			expectedValue: &expectedValueWithPrefix,
-		},
-		{
-			name: "both valueFromString and valueFromEnv is set",
-			data: []byte(`
-name: name
-valueFromEnv: 
-  name: ENVNAME
-value: value
-`),
-			wantErr: true,
+			name: "empty headers",
+			hs:   httpHeaders{},
+			want: map[string]string{},
 		},
 		{
-			name:    "none of valueFromString or valueFromEnv is set",
-			data:    []byte("name: name"),
-			wantErr: true,
+			name: "multiple headers",
+			hs:   httpHeaders{{Name: "header1", Value: &headerValue}, {Name: "header2", Value: &headerValue2}},
```

Comment:
What I'm missing in these test is the whole path from YAML through viper into the structure. (Previously, the tests had the same problem and thus didn't catch the bug you're fixing in this PR.)

I suggest the tests have YAML in them, then something like the following is used to fill HttpHeaders:
```go
	var s struct { HH httpHeaders } // the structure we are filling
	in := `hh: [...]`
	v := viper.New()
	v.SetConfigType("yaml")
	if err := v.ReadConfig(strings.NewReader(in)); err != nil {
		panic(err)
	}
	if err := v.UnmarshalExact(&s); err != nil {
		panic(err)
	}
```

Reply:
previous test had problem because of usage of custom unmarshaller. Unmarshaling yaml into struct using viper is already done by viper tests https://github.com/spf13/viper/blob/master/viper_test.go#L506

Reply:
That's what I'm saying: The tests did not verify that the values got from the YAML all the way to the structure using viper. (Think unit test.)

These tests have the same problem: They currently test that once the data is in httpHeaders, it gets to the map[string]string output as expected. (Think unit test.) There is no test that verifies the YAML to httpHeaders path -- httpHeader values in the tests are all literals.

In other words, the current tests are prone to essentially the same problem we hit previously.

What I'm suggesting is to test that YAML input, when processed by viper, produces the map[string]string output as expected. (Think integration test.)

It should be a mechanical change, with the little helper function I have provided.

Reply:
fixed b25af1336dbebc524011d8161541652b759b1788